cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -128,7 +120,13 @@ impl Cmd {
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
-                self.save_contract_id(&contract)?;
+                alias::Data::save_contract_id(
+                    &self.config.config_dir()?,
+                    &contract,
+                    self.alias.as_ref(),
```

Comment:
```suggestion
                    &self.alias,
```

Reply:
Tried this, but clippy complains about it, so I applied the suggestion:

```
error[E0308]: mismatched types
   --> cmd/soroban-cli/src/commands/contract/deploy/wasm.rs:126:21
    |
123 |                 alias::Data::save_contract_id(
    |                 ----------------------------- arguments to this function are incorrect
...
126 |                     &self.alias,
    |                     ^^^^^^^^^^^ expected `Option<&String>`, found `&Option<String>`
    |
    = note:   expected enum `std::option::Option<&std::string::String>`
            found reference `&std::option::Option<std::string::String>`
note: associated function defined here
   --> cmd/soroban-cli/src/commands/config/alias.rs:47:12
    |
47  |     pub fn save_contract_id(
    |            ^^^^^^^^^^^^^^^^
...
50  |         alias: Option<&String>,
    |         ----------------------
help: try using `.as_ref()` to convert `&std::option::Option<std::string::String>` to `std::option::Option<&std::string::String>`
    |
126 -                     &self.alias,
126 +                     self.alias.as_ref(),
    |

For more information about this error, try `rustc --explain E0308`.
error: could not compile `soroban-cli` (lib) due to 1 previous error
```